app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java

```diff
@@ -148,9 +162,105 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
             }
         }
         requestNotificationPermission();
+        scheduleWorkers();
         onNewIntent(getIntent());
     }
 
+    private void scheduleWorkers() {
+        WorkManager.getInstance(this)
+                .enqueueUniquePeriodicWork(
+                        AutoUpdateDescriptorsWorker.UPDATED_DESCRIPTORS_WORK_NAME,
+                        ExistingPeriodicWorkPolicy.KEEP,
+                        new PeriodicWorkRequest.Builder(AutoUpdateDescriptorsWorker.class, 24, TimeUnit.HOURS)
+                                .setConstraints(
+                                        new Constraints.Builder()
+                                                .setRequiredNetworkType(NetworkType.CONNECTED)
+                                                .build()
+                                ).build()
+                );
+        // TODO(aanorbel): add rules before checking updates
```

Comment:
which rules?

Reply:
Control for checking updates less frequently.